client/src/components/Login.js

```diff
@@ -168,11 +168,18 @@ const Login = (props) => {
           setLoading(false)
         }
       } catch (e) {
-        console.error(e)
-        setErrorMessage(
-          'Could not log in. Please check your username and password or contact the admin.'
-        )
-        setLoading(false)
+        console.error("Undefined User error:", e)
+        if (e.response.status === 401) {
```

Comment:
This logic is not quite right – 401 is used for all 'failed to log in' responses.
A better option would be to include the error message in the response from the server in all cases (as you have done for the new error).
If no error message is included, we could fall back to the generic message.

Reply:
I think I got it. Thanks for detailed feedback!